src/main/java/com/metsci/laproc/uicomponents/GraphExporter.java

```diff
@@ -0,0 +1,87 @@
+package com.metsci.laproc.uicomponents;
+
+import com.metsci.glimpse.layout.GlimpseLayoutProvider;
+import com.metsci.glimpse.painter.decoration.LegendPainter;
+import com.metsci.glimpse.painter.info.CursorTextPainter;
+import com.metsci.glimpse.painter.plot.XYLinePainter;
+import com.metsci.glimpse.painter.shape.PolygonPainter;
+import com.metsci.glimpse.plot.SimplePlot2D;
+import com.metsci.glimpse.support.color.GlimpseColor;
+import com.metsci.laproc.plotting.*;
+import com.metsci.laproc.utils.IAction;
+
+/**
+ * Provides a layout
+ * Created by malinocr on 1/21/2017.
+ */
+public class GraphExporter implements GlimpseLayoutProvider {
+    private Graph graph;
+
+    /**
+     * Constructor for a graph exporter
+     */
+    public GraphExporter(Graph graph) {
+        this.graph = graph;
+    }
+    /**
+     * Gets the layout for the graph
+     * @return GlimpseLayout layout for the graph
+     */
+    public SimplePlot2D getLayout()
+    {
+        // Create a plot frame
+        SimplePlot2D plot = new SimplePlot2D( );
+
+        Axis xAxis = graph.getXAxis();
+        Axis yAxis = graph.getYAxis();
+
+        // Set axis labels and chart title
+        plot.setTitle(graph.getTitle());
+        plot.setAxisLabelX(xAxis.getName());
+        plot.setAxisLabelY(yAxis.getName());
+
+        // Set the x, y initial axis bounds
+        plot.setMinX(xAxis.getMin());
+        plot.setMaxX(xAxis.getMax());
+
+        plot.setMinY(yAxis.getMin());
+        plot.setMaxY(yAxis.getMax());
+
+        // Only show the x and y crosshairs
+        plot.getCrosshairPainter().setVisible(false);
+
+        //Set up Legend
+        LegendPainter.LineLegendPainter legend = new LegendPainter.LineLegendPainter(LegendPainter.LegendPlacement.SE);
+        legend.setOffsetY(10);
+        legend.setOffsetX(100);
+        legend.setLegendItemWidth(60);
+
+        //Sets up all possible line colors for graphable data
+        float[][] possibleColors = new float[8][4];
```

Comment:
It seems like setting up this array each time the function is called is unnecessary. Can this be a field rather than a local variable?